examples/platform/silabs/efr32/rs911x/hal/efx_spi.c

```diff
@@ -322,13 +297,17 @@ sl_status_t sl_wfx_host_post_bootloader_spi_transfer(void)
 #if SL_LCDCTRL_MUX
 sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)
 {
+    uint32_t status = SL_STATUS_OK;
 #if SL_SPICTRL_MUX
-    if (sl_wfx_host_spi_cs_deassert() != SL_STATUS_OK)
-    {
-        return SL_STATUS_FAIL;
-    }
     xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY);
 #endif // SL_SPICTRL_MUX
+    status = sl_board_enable_display();
+    if (SL_STATUS_OK != status)
```

Comment:
this function is messy. The semaphore take is wrapped in `#if SL_SPICTRL_MUX` but not the give. Also why so many return points for essentially the same outcome?

```
sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)
{
      xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY);
      sl_status_t status = sl_board_enable_display();
      if (SL_STATUS_OK  == status)
      {
           status = sl_memlcd_refresh(sl_memlcd_get());
      }
      
      xSemaphoreGive(spi_sem_sync_hdl);
      return status;
}
```
Isn't that more readable and straightforward?
I also don't think you need a log for every error location. The return code should be telling enough for the caller to know what failed. 

I guess you can add logs if you want but avoid repetitive return points and be consistent with #if wrappers